settings.gradle

```diff
@@ -6,21 +6,23 @@ apply from: "${rootDir}/gradle/scripts/settings-flags.gradle"
 includeWithFlags ':bom',                       'bom'
 
 // Published Java projects
-includeWithFlags ':core',                      'java', 'publish', 'shade', 'trim'
-includeWithFlags ':grpc',                      'java', 'publish', 'relocate'
-includeWithFlags ':jetty',                     'java', 'publish', 'relocate'
-includeWithFlags ':kafka',                     'java', 'publish', 'relocate'
-includeWithFlags ':logback',                   'java', 'publish', 'relocate'
-includeWithFlags ':retrofit2',                 'java', 'publish', 'relocate'
-includeWithFlags ':spring-boot:autoconfigure', 'java', 'publish', 'relocate'
-includeWithFlags ':spring-boot:starter',       'java', 'publish', 'relocate'
-includeWithFlags ':testing',                   'java', 'publish', 'relocate'
-includeWithFlags ':thrift',                    'java', 'publish', 'relocate'
-includeWithFlags ':thrift0.9',                 'java', 'publish', 'relocate', 'no_aggregation'
-includeWithFlags ':tomcat',                    'java', 'publish', 'relocate'
-includeWithFlags ':tomcat8.0',                 'java', 'publish', 'relocate', 'no_aggregation'
-includeWithFlags ':zipkin',                    'java', 'publish', 'relocate'
-includeWithFlags ':zookeeper',                 'java', 'publish', 'relocate'
+includeWithFlags ':core',                       'java', 'publish', 'shade', 'trim'
+includeWithFlags ':grpc',                       'java', 'publish', 'relocate'
+includeWithFlags ':jetty',                      'java', 'publish', 'relocate'
+includeWithFlags ':kafka',                      'java', 'publish', 'relocate'
+includeWithFlags ':logback',                    'java', 'publish', 'relocate'
+includeWithFlags ':retrofit2',                  'java', 'publish', 'relocate'
+includeWithFlags ':spring:boot-autoconfigure',  'java', 'publish', 'relocate'
```

Comment:
I cannot find/remember the reference but because gradle cannot handle the projects that have same project name, need to rename a project using e.g. `project(':spring:boot:autoconfigure').name = 'boot-autoconfigure'`. To make less gradle configuration, changed directory structure to flatten spring-boot and spring-boot2 projects to make each project name unique.
(hope @anuraaga remember about such kind of gradle issue)

Reply:
I feel as if that issue should have been resolved by now, though I have seen plugins that don't use a project path and therefore can fail (gradle-protobuf-plugin) . Just wondering, did you try it and it didn't work? I guess it might work with Gradle, but not IntelliJ, which is why this sort of hack could be used instead

https://github.com/curioswitch/curiostack/blob/master/tools/gradle-plugins/gradle-curiostack-plugin/src/main/java/org/curioswitch/gradle/plugins/curiostack/CuriostackPlugin.java#L374

Flattening seems fine to me too though

Reply:
I see. Thanks for explanation. Actually, I tried but looks our thrift gradle didn't generate correctly with Gradle.